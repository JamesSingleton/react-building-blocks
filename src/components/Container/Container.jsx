import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledContainer = styled.div`
    padding-left: ${(props) => {
    if (props.full) return 0;
    return 'calc((100vw - 960px) / 2)';
  }};
    padding-right: ${(props) => {
    if (props.full) return 0;
    return 'calc((100vw - 960px) / 2)';
  }};
    padding-top: ${(props) => {
    if (props.fullVertical) return 0;
    if (props.small) return '15px';
    return '25px';
  }};
    padding-bottom: ${(props) => {
    if (props.fullVertical) return 0;
    if (props.small) return '15px';
    return '25px';
  }};
`;

const Container = ({
  fullVertical, full, small, children,
}) => (
  <StyledContainer
    fullVertical={fullVertical}
    full={full}
    small={small}
  >
    {children}
  </StyledContainer>
);

Container.propTypes = {
  fullVertical: PropTypes.bool,
  full: PropTypes.bool,
  small: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.element,
  ]).isRequired,
};

Container.defaultProps = {
  fullVertical: false,
  full: false,
  small: false,
};

export default Container;
