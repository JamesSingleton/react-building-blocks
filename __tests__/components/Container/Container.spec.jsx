import React from 'react';
import { mount } from 'enzyme';
import Container from '../../../src/components/Container';

describe('Container', () => {
  it('should render without crashing with just a child passed', () => {
    const renderedContainer = mount(<Container>Hello Span</Container>);
    expect(renderedContainer).toMatchSnapshot();
  });
  it('renders a small Container component', () => {
    const props = {
      small: true,
    };
    const renderedContainer = mount(<Container {...props}>Hello Small Container</Container>);
    expect(renderedContainer).toMatchSnapshot();
  });
  it('renders a full vertical Container component', () => {
    const props = {
      fullVertical: true,
    };
    const renderedContainer = mount(<Container {...props}>Full Vertical Container</Container>);
    expect(renderedContainer).toMatchSnapshot();
  });
  it('renders a full Container component', () => {
    const props = {
      full: true,
    };
    const renderedContainer = mount(<Container {...props}>Full Container</Container>);
    expect(renderedContainer).toMatchSnapshot();
  });
});
