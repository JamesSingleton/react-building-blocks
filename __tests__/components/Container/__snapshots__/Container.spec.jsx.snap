// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders a full Container component 1`] = `
<Container
  full={true}
  fullVertical={false}
  small={false}
>
  <styled.div
    full={true}
    fullVertical={false}
    small={false}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "dYLZGK",
            "rules": Array [
              "
    padding-left: ",
              [Function],
              ";
    padding-right: ",
              [Function],
              ";
    padding-top: ",
              [Function],
              ";
    padding-bottom: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      full={true}
      fullVertical={false}
      small={false}
    >
      <div
        className="sc-bdVaJa dYLZGK"
      >
        Full Container
      </div>
    </StyledComponent>
  </styled.div>
</Container>
`;

exports[`Container renders a full vertical Container component 1`] = `
<Container
  full={false}
  fullVertical={true}
  small={false}
>
  <styled.div
    full={false}
    fullVertical={true}
    small={false}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "lfOKPp",
            "rules": Array [
              "
    padding-left: ",
              [Function],
              ";
    padding-right: ",
              [Function],
              ";
    padding-top: ",
              [Function],
              ";
    padding-bottom: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      full={false}
      fullVertical={true}
      small={false}
    >
      <div
        className="sc-bdVaJa lfOKPp"
      >
        Full Vertical Container
      </div>
    </StyledComponent>
  </styled.div>
</Container>
`;

exports[`Container renders a small Container component 1`] = `
<Container
  full={false}
  fullVertical={false}
  small={true}
>
  <styled.div
    full={false}
    fullVertical={false}
    small={true}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "fOkvyZ",
            "rules": Array [
              "
    padding-left: ",
              [Function],
              ";
    padding-right: ",
              [Function],
              ";
    padding-top: ",
              [Function],
              ";
    padding-bottom: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      full={false}
      fullVertical={false}
      small={true}
    >
      <div
        className="sc-bdVaJa fOkvyZ"
      >
        Hello Small Container
      </div>
    </StyledComponent>
  </styled.div>
</Container>
`;

exports[`Container should render without crashing with just a child passed 1`] = `
<Container
  full={false}
  fullVertical={false}
  small={false}
>
  <styled.div
    full={false}
    fullVertical={false}
    small={false}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "bHeHMU",
            "rules": Array [
              "
    padding-left: ",
              [Function],
              ";
    padding-right: ",
              [Function],
              ";
    padding-top: ",
              [Function],
              ";
    padding-bottom: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      full={false}
      fullVertical={false}
      small={false}
    >
      <div
        className="sc-bdVaJa bHeHMU"
      >
        Hello Span
      </div>
    </StyledComponent>
  </styled.div>
</Container>
`;
